src/Commands/GenerateArchive.ts

```diff
@@ -97,7 +96,10 @@ export class GenerateArchive implements CommandModule {
     try {
       const parserFactory: ParserFactory = container.resolve(ParserFactory);
 
-      const parser: Parser = parserFactory.create(SpecType.NEXMOCK);
+      const parser = parserFactory.create(SpecType.NEXMOCK);
+      if (!parser) {
+        throw new Error('Incorrect a specification type');
+      }
```

Comment:
Let's move this condition to the ParserFactory implementation (DefaultParserFactory).